transport.go

```diff
@@ -59,9 +59,9 @@ func remapLocalHost(hostPorts []string) {
 }
 
 func parsePeer(peer string) (protocol, host string) {
-	// If we get a pure host:port, then we assume tchannel.
+	// If we get a pure host:port, we return unspecified to determine based on encoding
 	if _, _, err := net.SplitHostPort(peer); err == nil && !strings.Contains(peer, "://") {
-		return "tchannel", peer
+		return "unspecified", peer
```

Comment:
nit: "unspecified" should probably be a `const`, or maybe we should just use an empty string for protocol when it's unspecified. magic special values are a bit of a smell.

Reply:
+1